docs/adr/0006-rapid-releases-and-moving-towards-a-monorepo.md

```diff
@@ -0,0 +1,123 @@
+
+  * Good, because code will get merged and released more frequently.
+  * Good, because we don't need to spend time cutting releases.
+  * Good, because uplifts can be done much quicker.
+  * Good, because more of our release process will be automated.
+  * Bad, because we need to implement new automation.
+  * Bad, because when breaking changes cause the automatic nightly merge to fail for an application.  This means a developer will get the relatively urgent task of resolving things.  However, this can be mitigated by the policy of not merging changes until we're sure they can be taken by all the consumer applications.
+  * Good, because this avoids the current state of affairs where our `main` branch is not ready to be taken by the applications.  This means that a) our current code is not getting used and tested and b) it's very complicated to get other changes to that application, including bugfixes.
+  * Bad, because breaking changes require the corresponding PRs to be approved from all other teams before they can be merged into application-services.
+  * Good, because this is how things will work in a monorepo world.  Adjusting our workflow now prepares us for the future and will give us insights that may help guide the monorepo migration.
+  * Good, because our version numbers will align more closely with the rest of Mozilla.
+  * Good, because it makes it easier to know which application-services version the applications are using.  The current Firefox X release will always be using the latest application-services.X release -- for both the mobile apps and Desktop.
+  * Good, because we align better with the Firefox process and which may make it easier to use Mozilla tools later on, for example Buildhub or Product Details.
+
+# Non-normative discussion
+
+## Impact on other teams
+### Application teams -- Android, iOS, and Desktop
+
+* Phase 1
+  * The application-services version number style will change
+  * There may be a increase in the amount of application-service version upgrade PRs to approve.  We'll want to make sure that each application is on an application-services release that matches their release (i.e. avoiding having Fenix 110.x on application-services 109.x).  For the Android/iOS teams this will only be a small increase, but for Desktop it will be larger.
+* Phase 2
+  * More application-services PRs to approve, especially at the start.  Eventually, we should be able to arrange it so most PRs get auto-merged.
+
+### Release Management
+
+* Will now handle tagging, releasing, and version bumping application-services at the end of the nightly cycle.
+* Will now handle patch releases from the release / nightly branches
+
+### Release Engineering
+
+* Support the application-services team in developing the automation.  I believe most of the proposed changes are simple enough that application-services team can handle it ourselves.
```

Comment:
Just for the record, I know for sure Releng has this knowledge on Android. If I'm not mistaken another team dealt with this type of automation on Desktop (maybe the folks handling Lando?). I don't know who would be the right people to help with Firefox iOS. Because of the Apple ecosystem is mainly closed, the CI/CD is not centered on Taskcluster there.